mem/mem_solaris.go

```diff
@@ -34,6 +35,13 @@ func VirtualMemoryWithContext(ctx context.Context) (*VirtualMemoryStat, error) {
 			return nil, err
 		}
 		result.Total = cap
+		freemem, err := globalZoneFreeMemory()
```

Comment:
How about to pass `ctx` instead of creating a new context in this function?

```suggestion
		freemem, err := globalZoneFreeMemory(ctx)
```

Reply:
Hi Shirou, 
Thank for your review and comment; I have made the suggested change and submitted the commit. 
For the tested OS environments, below is what I did for the Solaris and illumos environments:
1. For the Solaris OS test, I have created the Solaris instance from the Oracle Solaris 11.4 image provided by the Oracle Cloud marketplace at https://cloudmarketplace.oracle.com/marketplace/en_US/listing/61750333 (OCI allows to create free Solaris instances with the public image);
2. For the illumos OS test, I used the omniOS-r151044 (recommended by the [illumos.org](http://illumos.org/)) from https://omnios.org/download.html to create the virtual machine and do the test. 
For both environments, I tested with the go1.15  illumos/amd64 (and go1.13 illumos/amd64) and tests result look good.